src/test/java/ch/uzh/ifi/hase/soprafs22/service/UserServiceTest.java

```diff
@@ -165,9 +263,413 @@ void updateUser_Conflict(){ //Test what happens when user tries to change userna
+
+        Match testMatch = userService.createMatch(otherUser, testUser2);
+        testMatch.setMatchId(2L);
+
+        Mockito.when(matchRepository.findByMatchId(2L)).thenReturn(testMatch);
+        userService.setMatch(testMatch);
+        assertFalse(userService.doesMatchExist(testUser, otherUser));
+        assertFalse(userService.doesMatchExist(otherUser, testUser));
+    }
+
+    @Test
+    void checkGeneralAccess_granted(){
+      testUser.setToken("token");
+      Mockito.when(userRepository.findByToken(testUser.getToken())).thenReturn(testUser);
+      userService.checkGeneralAccess(testUser.getToken());
+      Mockito.verify(userRepository, Mockito.times(1)).findByToken(Mockito.any());
+    }
+
+    @Test
+    void checkGeneralAccess_denied(){
+        testUser.setToken("token");
+        Mockito.when(userRepository.findByToken(testUser.getToken())).thenReturn(null);
+        ResponseStatusException e = assertThrows(ResponseStatusException.class, ()-> userService.checkGeneralAccess("token"));
+        assertEquals(HttpStatus.UNAUTHORIZED, e.getStatus());
+    }
+
+    @Test
+    void getCurrentBlackCards_emptyListReturned(){
+      Mockito.when(userRepository.findById(1L)).thenReturn(testUser);
+      assertEquals(Collections.emptyList(), userService.getCurrentBlackCards(testUser.getId()));
+      testUser.setUserBlackCards(null);
+      assertEquals(Collections.emptyList(), userService.getCurrentBlackCards(testUser.getId()));
+      testUser.setUserBlackCards(new UserBlackCards());
+      assertEquals(Collections.emptyList(), userService.getCurrentBlackCards(testUser.getId()));
+      List<BlackCard> cards = new ArrayList<>();
+      UserBlackCards ubc = new UserBlackCards();
+      ubc.setBlackCards(cards);
+      assertEquals(Collections.emptyList(), userService.getCurrentBlackCards(testUser.getId()));
+      testUser.setUserBlackCards(userBlackCards);
+      // ToDo: for 100% line coverage: test with Powermock to access private final Date of userBlackCards
+      //assertEquals(Collections.emptyList(), userService.getCurrentBlackCards(testUser.getId()));
+    }
+
+    @Test
+    void getCurrentBlackCards_blackCardReturned() {
+        Mockito.when(userRepository.findById(1L)).thenReturn(testUser);
+        testUser.setUserBlackCards(userBlackCards);
+        List<BlackCard> expected = new ArrayList<>();
+        expected.add(testBlackCard);
+        List<BlackCard> actual = userService.getCurrentBlackCards(testUser.getId());
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void assignBlackCardsToUser_successSingleCard() {
+        Mockito.when(userRepository.findById(1L)).thenReturn(testUser);
+        List<BlackCard> cards = new ArrayList<>();
+        cards.add(testBlackCard);
+        userService.assignBlackCardsToUser(testUser.getId(), cards);
+        UserBlackCards ubc = testUser.getUserBlackCards();
+        assertArrayEquals(new List[]{cards}, new List[]{ubc.getBlackCards()});
```

Comment:
Why  do you use assertArrayEquals?

Reply:
as the list only has one element it is correct that it is not needed